### PR TITLE
Artist show page

### PR DESCRIPTION
**Pull request by R-Aranda**
All bullets for artist show page:
-Created show page component container
-Created a "header" component for the ArtistShowPage.js component. Once we create a reviews index component, we will add that component to the ArtistShowPage.js container.
-Added show method to the artist controller
-Added show page path to react router
-Added some foundation and styling  to show page
-Added controller testing for show page

Notes:
We did not update the routes.rb file to explicitly "allow" specific paths because that has been done on a future index page PR (deleting `get "*path", to:"homes#index", via: :all`). Once that PR has been merged, we will use that routes.rb file.

### CODE REVIEW

File: app/javascript/react/components/Artists/ArtistShowPage.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      }, []); 
+
+    return ( 
+        <div>
+            <ArtistTopSection
+            key={artist.id}
+            name={artist.name}
+            bio={artist.bio}
+            genre={artist.genre}
+            website={artist.website}
```
Comment (nwalberts) [inline app/javascript/react/components/Artists/ArtistShowPage.js]: convention is to indent all props in. Indent in lines 30 to 34

File: app/javascript/react/components/Artists/ArtistTopSection.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+const ArtistTopSection = ({ name, bio, website, genre}) => {
+    return (  
+        <div className="news-image-gallery-container">
+
+            <div className="row">
+                <div className="small-12 medium-12 large-4 row">
+                    <div className="rounded-social-buttons">
+                        <div className='container'>
+                            <img className="img"src='https://images.unsplash.com/photo-1506755855567-92ff770e8d00?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=774&q=80'/>
+                        </div>
```
Comment (nwalberts) [inline app/javascript/react/components/Artists/ArtistTopSection.js:12]: unless you plan on having this adorable kitten image in your website forever, remove this until you decide on something more permanent!

File: app/javascript/react/components/Artists/ArtistTopSection.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+                     <a className="social-button facebook" href="#"><i className="fab fa-facebook"/></a>
+                     <a className="social-button twitter" href="#"><i className="fab fa-twitter"/></a>
+                     <a className="social-button spotify" href="#"><i className="fab fa-spotify"/></a>
+                     <div className="clearfix"></div>
+                            <h4 className="news-image-gallery-title">{name}</h4>
+                            <p>{bio}</p>
+                            <p> <a href="#" className="read-more"> Read More on their website </a> </p>
```
Comment (nwalberts) [inline app/javascript/react/components/Artists/ArtistTopSection.js]: hmmm, this link to this website will actually have to be REAL. and working. Update your seeds file to seed artists that exist with real website links. Then, make sure that this is not hard-coded to go nowhere

File: app/javascript/react/components/Artists/ArtistTopSection.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+                            <p>Genre: {genre}</p>
+                            <div>
+                                <p> Average Rating: /5 </p>
```
Comment (nwalberts) [inline app/javascript/react/components/Artists/ArtistTopSection.js]: I would remove this average rating for now. Its technically an unfinished feature. You can always add it in later

File: config/routes.rb [hunk 5]
```diff
@@ -8,7 +8,9 @@
   
   namespace :api do
     namespace :v1 do
-      resources :artists, only: [:index]
+      resources :artists, only: [:index, :show]
     end
   end
+
+  get "*path", to:"homes#index", via: :all
```
Comment (nwalberts) [inline config/routes.rb]: as mentioned earlier, I dont want folks using the wild-card path. Instead, for any new paths you should add, you should be adding those explicitly

File: db/seeds.rb [hunk 6]
```diff
@@ -5,3 +5,10 @@
 #
 #   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
 #   Character.create(name: 'Luke', movie: movies.first)
+
+
+Artist.create(name: "Daft Punk", bio: "Two French men making crazy electronic music")
+Artist.create(name: "Kendrick Lamar", bio: "Compton's second coming of Christ")
+Artist.create(name: "Pearl Jam", bio: "Eddie Vedder the GOAT! Nirvana who? This is the only Seattle band that matters")
+Artist.create(name: "Nirvana", bio: "Kurt Cobain the GOAT! Pearl Jam who? This is the only Seattle band that matters")
+Artist.create(name: "Hole", bio: "Good band BUT I HATE COURTNEY LOVE!!!")
```
Comment (nwalberts) [inline db/seeds.rb]: dope. Add their actual websites as well!

File: spec/controllers/api/v1/artists_controller_spec.rb [hunk 7]
```diff
@@ -27,4 +27,16 @@
       expect(returned_json[1]["bio"]).to  eq second_artist.bio
     end
   end 
+
+  describe "GET#index" do
```
Comment (nwalberts) [inline spec/controllers/api/v1/artists_controller_spec.rb]: uh oh, this says get INDEX, but this tests is for the show

**Review by nwalberts — CHANGES_REQUESTED** [review]

**Review by devinrojas — COMMENTED** [review]
-Updated and cleaned up all the comments you left
-Added websites to all of the seeded artists
-indented props in our  "Artist Top Section" 
-Improved controller test to have it GET the show page

**Pull request merged**